### PR TITLE
Update doc for CA1034 to mention builder pattern

### LINKED ISSUE #4994: CA1034 - Allow builder pattern
The rule was updated to allow the builder pattern. It is now allowed to have an inner type whose name ends in `Builder` and whose outer class has no publicly visible (public or protected) constructor. The containing type should not be nested.

See https://github.com/dotnet/roslyn-analyzers/pull/3318

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d124eb75-6775-f608-06cc-cbe2ef0dddf8
* Version Independent ID: 93507687-1962-6e38-1b5a-8794e579ecb0
* Content: [CA1034: Nested types should not be visible - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1034?view=vs-2019)
* Content Source: [docs/code-quality/ca1034.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/ca1034.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-code-analysis**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #4994

Note that this exclusion is not yet publicly released, you might want to sync with @mavasani on when this PR can be merged.

### CODE REVIEW

**Comment by PRMerger13** [comment]
@Evangelink : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/code-quality/ca1034.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ ms.workload:
 
 ## Cause
 
-An externally visible type contains an externally visible type declaration. Nested enumerations and protected types are exempt from this rule.
+An externally visible type contains an externally visible type declaration. Nested enumerations, protected types and builder patterns are exempt from this rule.
```
Comment (Evangelink) [inline docs/code-quality/ca1034.md]: Do you want to explain what is the builder pattern? Do you want to list the exact allowance conditions?

Comment (mavasani) [reply docs/code-quality/ca1034.md:33]: I think it is probably intuitive enough. We can add it if we get feedback about it not being clear.

**Comment by Mikejo5000** [comment]
#sign-off

**Comment by PRMerger13** [comment]
@ktoliver : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Pull request merged**